micrometer-core/src/main/java/io/micrometer/core/instrument/AbstractTimer.java

```diff
@@ -258,7 +258,7 @@ public void record(Runnable f) {
     @Override
     public final void record(long amount, TimeUnit unit) {
         if (amount >= 0) {
-            histogram.recordLong(TimeUnit.NANOSECONDS.convert(amount, unit));
+            histogram.recordLong(baseTimeUnit().convert(amount, unit));
```

Comment:
Please remove this change, this should be recorded as nanos and converted to the right unit at publish-time.